qemu/domain.go

```diff
@@ -372,6 +372,14 @@ func (d *Domain) Events() (chan qmp.Event, chan struct{}, error) {
 	// handle disconnection
 	go func() {
 		<-done
+		// drain anything that gets sent on the channel
+		// because the disconnect won't be processed if the
+		// listenAndServe loop is waiting for the listener
+		// to read from the unbuffered channel.
+		go func() {
+			for range stream {
+			}
```

Comment:
nit: the newline isn't needed, `for range stream{}` would do.

Reply:
apparently go fmt is making that newline.